Makefile

```diff
@@ -112,7 +118,7 @@ test: test-unit ## Run the tests
 test-markdown test/markdown:
 	./hack/ci/marker
 
-test-sanity test/sanity: tidy
+test-sanity test/sanity: tidy install
```

Comment:
I don't think we should be modifying state outside of the local dir unless the user explicitly asks for it. The `sanity-check.sh` already contains `go vet ./...`, and if you want to test compilation I suggest creating a target for `go build ./...` which changes no state (except for potentially changing go.mod or go.sum).

Reply:
This isn't to test compilation. The `operator-sdk` binary is now required by the sanity test to run the `generate` subcommands that the sanity test calls in the hack script.

Making `install` a prerequisite of `test-sanity` was an original concern of mine as well. We could build a local copy and put it in `build/`, maybe?

Reply:
Yes I'd rather run `make build/operator-sdk`.

Reply:
Hi @estroz, 

This suggestion will not work. Note that the make build/operator-sdk do not build a bin which allow us use the command.

Note that:

```shell
$ make build/operator-sdk
make: `build/operator-sdk' is up to date.
camilamacedo@Camilas-MacBook-Pro ~/go/src/github.com/operator-framework/operator-sdk (fix-metrics-local) $ operator-sdk version
-bash: operator-sdk: command not found
```

```
$ go build ./...
camilamacedo@Camilas-MacBook-Pro ~/go/src/github.com/operator-framework/operator-sdk (fix-metrics-local) $ operator-sdk version
-bash: operator-sdk: command not found

```

Reply:
Hi @estroz @joelanford, 

Have you any idea or can we just keep the install?